server/api/smallgroup/smallgroup.controller.js

```diff
@@ -132,6 +132,20 @@ exports.addMember = function(req, res){
 
 };
 
+exports.changeName = function(req,res){
+  var id = req.params.id;
+  var newName = String(req.body.smallGroupName);
+  SmallGroup.findById(id, function(err,smallgroup){
+    if (err) return handleError(res, err);
+    smallgroup.name = newName;
+    res.json({name:smallgroup.name});
+    smallgroup.save(function(err){
+      if (err) return validationError(res,err);
+      res.send(200);
```

Comment:
This causes an error on the server - line 141 (`res.json({name:smallgroup.name});`) should happen after saving (instead of the current line 144)  It should also be returned ( i.e., `return res.json. ..... `)